File: 08_student_supports/java/src/Main.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        System.out.println("\n-- PROGRAMA PARA ENTREGAR SUBSIDIOS A ESTUDIANTES --\n");

        System.out.println("Ingrese el estrado del estudiante:");
        int estrato = input.nextInt();

        if (estrato <= 0) {
            System.err.println("\nEl estrato no puede ser negativo o cero");
            return;
        } else if (estrato > 6) {
            System.err.println("\nEl estrato no puede ser mayor a 6");
            return;
        }

        System.out.println("Ingrese el promedio del estudiante:");
        double average = input.nextDouble();

        if (average < 0) {
            System.err.println("\nEl promedio no puede ser negativo");
            return;
        } else if (average > 5) {
            System.err.println("\nEl promedio no puede ser mayor a 5");
            return;
        }

        int subsidio = 0;

        if ((estrato == 1 || estrato == 2) && average >= 4.0) {
            subsidio = 100000;
        } else if (estrato == 3 && average > 4.0) {
            subsidio = 80000;
        } else if (estrato > 3 && average > 4.5) {
            subsidio = 50000;
        } else {
            System.out.println("\nEl estudiante no cumple con las condiciones para recibir un subsidio");
            return;
        }

        System.out.println("\nEl subsidio para el estudiante es: $ " + subsidio);
    }
}

File: 08_student_supports/ocaml/bin/main.ml
let () = print_endline "\n-- PROGRAMA PARA ENTREGAR SUBSIDIOS A ESTUDIANTES --\n"

let social_class = print_endline "Ingrese el estrato del estudiante:"; int_of_string (read_line ())

let () =
  if social_class >= 1 && social_class <= 6 then
      let grade = prerr_endline "Ingrese el promedio del estudiante:"; float_of_string (read_line ()) in

      if grade >= 0. && grade <= 5. then
        let support =
          if (social_class = 1 || social_class = 2) && grade >= 4. then
            100000
          else if social_class = 3 && grade > 4. then
            80000
          else if social_class > 3 && grade > 4.5 then
            50000
          else
            0 in
            
        if support > 0 then
          print_endline (Printf.sprintf "\nEl subsidio para el estudiante es: $ %d" support)
        else
          print_endline "\n El estudiante no cumple con las condiciones para recibir un subsidio"

      else
        prerr_endline "\nEl promedio debe ser un número entre 0 y 5"
  else
      prerr_endline "\nEl estrato debe ser un número entre 1 y 6"